### ISSUE TITLE
upsertGraph, should update root object when relational change

### ISSUE THREAD

**Issue by johannessjoberg** [Vincit/objection.js#523]
When passing an object to `upsertGraph` with only id and relational property, the relational property is not updated. But when adding an extra property which is not relational, this property and the relational property is updated.

I have boiled it down to that this is because the relational property is not considered as a "change" on it's own.

Below are four tests which can verify this behaviour. Is this expected behaviour?

Happy to supply a PR if suitable! Thanks for a great repo.

Objection.js 0.8.9
For verification purposes, add these to:`tests/integration/upsertGraph.js`
```
it('should update belongsToOneRelation with only relational property => fails', () => {
      const upsert = {
        id: 1,
        // belongsToOne: From null => 3
        model1Relation1: { id: 3 },
      }

      return transaction(session.knex, trx => {
        return Model1
          .query(trx)
          .upsertGraph(upsert, { relate: true, unrelate: true })
          .then(result => {
            // Fetch the graph from the database.
            return Model1
              .query(trx)
              .findById(result.id)
              .eager('model1Relation1')
              .select('id')
              .modifyEager('model1Relation1', qb => qb.select('id'));
          })
          .then(omitIrrelevantProps)
          .then(result => {
            expect(result).to.eql({
              id: 1,
              model1Relation1: {
                id: 3,
              }
            });
          });
      }); 
    });

    it('should update belongsToOneRelation with model1Prop1 and relational property => works', () => {
      const upsert = {
        id: 1,
        model1Prop1: 'updated',
        // belongsToOne: From null => 3
        model1Relation1: { id: 3 },
      }

      return transaction(session.knex, trx => {
        return Model1
          .query(trx)
          .upsertGraph(upsert, { relate: true, unrelate: true })
          .then(result => {
            // Fetch the graph from the database.
            return Model1
              .query(trx)
              .findById(result.id)
              .eager('model1Relation1')
              .select('id')
              .modifyEager('model1Relation1', qb => qb.select('id'));
          })
          .then(omitIrrelevantProps)
          .then(result => {
            expect(result).to.eql({
              id: 1,
              model1Relation1: {
                id: 3,
              }
            });
          });
      }); 
    });

    it('should update belongsToOneRelation with only new relation property => fails', () => {
      const upsert = {
        id: 1,
        // belongsToOne: From null => new
        model1Relation1: { model1Prop1: 'new' },
      }

      return transaction(session.knex, trx => {
        return Model1
          .query(trx)
          .upsertGraph(upsert, { relate: true, unrelate: true })
          .then(result => {
            // Fetch the graph from the database.
            return Model1
              .query(trx)
              .findById(result.id)
              .eager('model1Relation1')
              .select('id')
              .modifyEager('model1Relation1', qb => qb.select('model1Prop1'));
          })
          .then(omitIrrelevantProps)
          .then(result => {
            expect(result).to.eql({
              id: 1,
              model1Relation1: {
                model1Prop1: 'new',
              }
            });
          });
      }); 
    });

    it('should update belongsToOneRelation with model1Prop1 and new relation property => works', () => {
      const upsert = {
        id: 1,
        model1Prop1: 'updated',
        // belongsToOne: From null => new
        model1Relation1: { model1Prop1: 'new' },
      }

      return transaction(session.knex, trx => {
        return Model1
          .query(trx)
          .upsertGraph(upsert, { relate: true, unrelate: true })
          .then(result => {
            // Fetch the graph from the database.
            return Model1
              .query(trx)
              .findById(result.id)
              .eager('model1Relation1')
              .select('id')
              .modifyEager('model1Relation1', qb => qb.select('model1Prop1'));
          })
          .then(omitIrrelevantProps)
          .then(result => {
            expect(result).to.eql({
              id: 1,
              model1Relation1: {
                model1Prop1: 'new',
              }
            });
          });
      }); 
    });
```

**Comment by koskimas** [comment]
Thank you for an excellent issue! Seems like a clear bug. A PR would be awesome!

**Comment by koskimas** [comment]
Fixed by @johannessjoberg in ced27b7eab1d64ef74e3fe4fc4fc816badca31ec
